crypto/hmac/hmac.c

```diff
@@ -37,7 +37,8 @@ int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
     if (key != NULL) {
         reset = 1;
         j = EVP_MD_block_size(md);
-        OPENSSL_assert(j <= (int)sizeof(ctx->key));
+        if (!ossl_assert(j <= (int)sizeof(ctx->key)))
+            goto err;
```

Comment:
More specifically EVP_MD_meth_set_input_blocksize could [if not should?] enforce some boundary condition. Of course it might turn out that the boundary condition won't be equivalent to sizeof(ctx->key), HMAC_MAX_MD_CBLOCK, defined as 128 in public hmac.h. In which case yes, it would have to be classified as run-time condition.